projects/js-packages/components/components/record-meter-donut/index.tsx

```diff
@@ -0,0 +1,104 @@
+import { __ } from '@wordpress/i18n';
+import React from 'react';
+import './style.scss';
+export type DonutMeterProps = {
+	/**
+	 * Total number of items for the donut meter.
+	 */
+	totalCount: number;
+	/**
+	 * Count for the given item
+	 */
+	segmentCount: number;
+	/**
+	 * Color code for the background color for the item.
+	 * If not provided, defaults to Jetpack Green
+	 */
+	backgroundColor?: string;
+
+	/**
+	 * thickness for the chart border
+	 * If not provided, defaults to 3.5
+	 */
+	thickness?: string;
+	/**
+	 * width for the full chart size
+	 * If not provided, defaults to 64px
+	 */
+	donutWidth?: string;
+};
+
+/**
+ * Generate record meter donut bar
+ *
+ * @param {DonutMeterProps} props - Props
+ * @returns {React.ReactElement} - JSX element
+ */
+const DonutMeter: React.FC< DonutMeterProps > = ( {
+	totalCount,
+	segmentCount,
+	backgroundColor = '#00BA37', // jetpack green fallback
+	thickness = '3.5',
+	donutWidth = '64px',
+} ) => {
+	const count = () => {
+		// get count as a percent value
+		return ( segmentCount / totalCount ) * 100;
+	};
+
```

Comment:
Should we add some bounds checking on our props here? For example, making sure the segmentCount is less than the totalCount, keeping thickness between 1 & some upper bound, setting a minimum donutWidth?

Reply:
Yes, absolutely! If it gets to thick, for example, it stops rounding out so nicely. It would be excellent to prevent that. And also checking the segmentCount and totalCount could be checked here for sure. The decision surrounding this would be deciding what to show if the segmentCount was, for example, over the totalCount. Depending on how the new billing API data comes through, it may be possible that someone is over their limit and we would want to display the chart just at 100%.